test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ViewFeatures/ViewDataDictionaryTest.cs

```diff
@@ -50,33 +49,46 @@ public void ConstructorInitalizesMembers()
         }
 
         [Fact]
-        public void SetModelUsesPassedInModelMetadataProvider()
+        public void Constructor_UsesModelMetadataProvider()
         {
             // Arrange
-            var metadataProvider = new Mock<IModelMetadataProvider>();
+            var metadataProvider = new Mock<IModelMetadataProvider>(MockBehavior.Strict);
             metadataProvider
                 .Setup(m => m.GetMetadataForType(typeof(object)))
                 .Returns(new EmptyModelMetadataProvider().GetMetadataForType(typeof(object)))
                 .Verifiable();
+            var modelState = new ModelStateDictionary();
+
+            // Act
+            var viewData = new ViewDataDictionary(metadataProvider.Object, modelState);
+
+            // Assert
+            Assert.NotNull(viewData.ModelMetadata);
+            metadataProvider.Verify(m => m.GetMetadataForType(typeof(object)), Times.Once());
+        }
+
+        [Fact]
+        public void SetModel_DoesNotUseModelMetadataProvider_IfTypeCompatible()
+        {
+            // Arrange
+            var metadataProvider = new Mock<IModelMetadataProvider>(MockBehavior.Strict);
             metadataProvider
                 .Setup(m => m.GetMetadataForType(typeof(TestModel)))
                 .Returns(new EmptyModelMetadataProvider().GetMetadataForType(typeof(TestModel)))
                 .Verifiable();
-            var modelState = new ModelStateDictionary();
-            var viewData = new TestViewDataDictionary(metadataProvider.Object, modelState);
+            var viewData = new TestViewDataDictionary(metadataProvider.Object, typeof(TestModel));
             var model = new TestModel();
 
             // Act
             viewData.SetModelPublic(model);
 
             // Assert
             Assert.NotNull(viewData.ModelMetadata);
-            metadataProvider.Verify();
+            metadataProvider.Verify(m => m.GetMetadataForType(typeof(TestModel)), Times.Once());
         }
 
-        // When SetModel is called, only GetMetadataForType from MetadataProvider is expected to be called.
         [Fact]
-        public void SetModelCallsGetMetadataForTypeExactlyOnce()
+        public void SetModel_UsesModelMetadataProvider_IfSourceTypeIsObject()
```

Comment:
Can we say something like "GetsNewMetadata" instead of "UsesModelMetadataProvider"?